astropy/cosmology/_src/tests/test_deprecated_modules.py

```diff
@@ -0,0 +1,101 @@
+"""Testing deprecated modules in `astropy.cosmology`."""
+
+import re
+import warnings
+
+import pytest
+
+
+def test_connect():
+    """Test `astropy.cosmology.connect`."""
+    from astropy.cosmology import connect
+
+    try:
+        del connect.CosmologyFromFormat
+    except Exception:
+        pass
+
+    with (
+        warnings.catch_warnings(),
```

Comment:
could you add a comment here (and in following similar tests) to explain why this filtering is needed ?